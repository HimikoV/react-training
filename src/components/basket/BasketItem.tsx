import {
  Box,
  Card,
  CardMedia,
  CardContent,
  Typography,
  IconButton,
} from "@mui/material";
import AddShoppingCartOutlinedIcon from "@mui/icons-material/AddShoppingCartOutlined";
import RemoveShoppingCartOutlinedIcon from "@mui/icons-material/RemoveShoppingCartOutlined";
const BasketItem = ({
  item,
  reduceBasketItemQuant,
  increaseBasketItemQuant,
}: any) => {
  return (
    <Card sx={{ m: 0.5 }}>
      <Box
        sx={{
          display: "flex",
          flexWrap: " wrap",
          color: "#3d405b",
          justifyContent: "space-between",
          alignItems: "center",
          flexDirection: "row",
          width: 700,
          minWidth: 500,
        }}
      >
        <Box
          sx={{
            display: "flex",
            flexWrap: "wrap",
            color: "#3d405b",
            justifyContent: "flex-start",
            alignItems: "center",
            flexDirection: "row",
          }}
        >
          <Box
            sx={{
              flexWrap: "wrap",
              color: "#3d405b",
              justifyContent: "flex-start",
              alignItems: "center",
              flexDirection: "row",
              width: 100,
            }}
          >
            <CardMedia
              component="img"
              height="100"
              image="https://images.unsplash.com/photo-1515471897120-85416077e011?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8Y29mZmVlJTIwYmFnfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=500&q=60"
              alt="green iguana"
            />
          </Box>

          <CardContent>
            <Typography
              sx={{ fontSize: 14 }}
              color="text.secondary"
              gutterBottom
            >
              {item.name}
            </Typography>
            <Typography variant="h6" component="div">
              price: {item.price}$
            </Typography>
          </CardContent>
        </Box>

        <Box
          sx={{
            display: "flex",
            flexWrap: "wrap",
            color: "#3d405b",
            justifyContent: "center",
            alignItems: "center",
            flexDirection: "row",
          }}
        >
          <IconButton
            onClick={() => {
              reduceBasketItemQuant(item.id);
            }}
          >
            <RemoveShoppingCartOutlinedIcon fontSize="large" />
          </IconButton>
          <Typography variant="h6" component="span">
            {item.quantity}
          </Typography>
          <IconButton
            onClick={() => {
              increaseBasketItemQuant(item.id);
            }}
          >
            <AddShoppingCartOutlinedIcon fontSize="large" />
          </IconButton>
        </Box>
      </Box>
    </Card>
  );
};
export default BasketItem;
