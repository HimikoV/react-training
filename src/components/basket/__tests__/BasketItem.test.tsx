import { describe, it } from "@jest/globals";
import renderer from "react-test-renderer";
import BasketItem from "../BasketItem";

describe("testing BasketItem component", () => {
  it("BasketItem Renders Correctly", () => {
    const basketItem = renderer
      .create(
        <BasketItem
          item={{
            name: "item 1",
            price: 20,
            category: "category_1",
            id: "1",
            quantity: 2,
          }}
          reduceBasketItemQuant={() => {}}
          increaseBasketItemQuant={() => {}}
          deleteBasketItem={() => {}}
        />
      )
      .toJSON();
    expect(basketItem).toMatchSnapshot();
  });
});
