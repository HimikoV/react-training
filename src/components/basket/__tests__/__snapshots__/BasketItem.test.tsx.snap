// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing BasketItem component BasketItem Renders Correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-t7yft3-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiGrid-root css-ahvzxn-MuiGrid-root"
  >
    <div
      className="MuiGrid-root css-2etixe-MuiGrid-root"
    >
      <img
        alt="green iguana"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="100"
        src="https://images.unsplash.com/photo-1515471897120-85416077e011?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8Y29mZmVlJTIwYmFnfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=500&q=60"
      />
    </div>
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <span
        className="MuiTypography-root MuiTypography-caption css-1d9s20a-MuiTypography-root"
      >
        item 1
      </span>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1invrju-MuiTypography-root"
      >
        price: 
        20
        $
      </h5>
    </div>
  </div>
  <div
    className="MuiGrid-root css-112f6m6-MuiGrid-root"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="RemoveShoppingCartOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M1.41 1.13 0 2.54l4.39 4.39 2.21 4.66-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h7.46l1.38 1.38c-.5.36-.83.95-.83 1.62 0 1.1.89 2 1.99 2 .67 0 1.26-.33 1.62-.84L21.46 24l1.41-1.41L1.41 1.13zM7 15l1.1-2h2.36l2 2H7zM20 4H7.12l2 2h9.19l-2.76 5h-1.44l1.94 1.94c.54-.14.99-.49 1.25-.97l3.58-6.49C21.25 4.82 20.76 4 20 4zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </button>
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1invrju-MuiTypography-root"
    >
      quant: 
      2
    </h5>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="AddShoppingCartOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="DeleteForeverOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.12 10.47 12 12.59l-2.13-2.12-1.41 1.41L10.59 14l-2.12 2.12 1.41 1.41L12 15.41l2.12 2.12 1.41-1.41L13.41 14l2.12-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4zM6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9z"
        />
      </svg>
    </button>
  </div>
</div>
`;
